opentelemetry-exporter-gcp-trace/lib/opentelemetry/exporter/google_cloud_trace/translator.rb

```diff
@@ -0,0 +1,240 @@
+        # exported.
+        # @return [Google::Cloud::Trace::V2::BatchWriteSpansRequest]
+        # The request message for the BatchWriteSpans method.
+        def create_batch spans
+          cloud_trace_spans = []
+
+          spans.each do |span|
+            trace_id = span.hex_trace_id
+            span_id = span.hex_span_id
+            parent_id = span.hex_parent_span_id
+            span_name = "projects/#{@project_id}/traces/#{trace_id}/spans/#{span_id}"
+            cloud_trace_spans << Google::Cloud::Trace::V2::Span.new(
+              name: span_name,
+              span_id: span_id,
+              parent_span_id: parent_id,
+              display_name: create_name(span.name, MAX_DISPLAY_NAME_BYTE_COUNT),
+              start_time: create_time(span.start_timestamp),
+              end_time: create_time(span.end_timestamp),
+              attributes: create_attributes(span.attributes, MAX_SPAN_ATTRIBUTES, add_agent_attribute: true),
+              links: create_links(span.links),
+              status: create_status(span.status),
+              time_events: create_time_events(span.events),
+              span_kind: create_span_kind(span.kind)
+            )
+          end
+          {
+            name: "projects/#{@project_id}",
+            spans: cloud_trace_spans
+          }
+        end
+
+        private
+
+        def create_time epoch
+          return if epoch.nil?
+          time_nsec = Time.at 0, epoch, :nsec # create Time from nanoseconds epoch
+          Google::Protobuf::Timestamp.from_time time_nsec
+        end
+
+        def create_name name, max_bytes
+          truncated_str, truncated_byte_count = truncate_str name, max_bytes
+          Google::Cloud::Trace::V2::TruncatableString.new value: truncated_str,
+                                                          truncated_byte_count: truncated_byte_count
+        end
+
+        def truncate_str str, max_bytes
+          encoded = str.encode "utf-8"
+          truncated_str = encoded.byteslice 0, max_bytes
+          [truncated_str, encoded.length - truncated_str.encode("utf-8").length]
+        end
+
+        def create_attributes attributes, max_attributes, add_agent_attribute: false
+          return if attributes.nil? || attributes.empty?
+          attribute_map = {}
+
+          if add_agent_attribute
+            attribute_map["g.co/agent"] = create_attribute_value(
+              "opentelemetry-ruby #{Gem.loaded_specs['opentelemetry-sdk'].version};" \
+              "google-cloud-trace-exporter #{OpenTelemetry::Exporter::GoogleCloudTrace::VERSION}"
+            )
+          end
+
+          attributes.each_pair do |k, v|
+            key = truncate_str(k.to_s, MAX_ATTRIBUTES_KEY_BYTES).first
+            key = LABELS_MAPPING[key.to_sym] if LABELS_MAPPING.key? key.to_sym
+            value = create_attribute_value v
+            attribute_map[key] = value unless value.nil?
+
+            break if attribute_map.count == max_attributes
```

Comment:
I'm paranoid: I tend to prefer `>=` rather than `==` for this kind of logic, just to prevent a future bug where we could start setting two attributes per iteration and we end up leapfrogging over `max_attributes`.